commit fb40cd9d914c8a1b4acc56edaa96046c5f4f1669
Author: Miri Korenblit <miriam.rachel.korenblit@intel.com>
Date:   Thu Apr 13 21:40:28 2023 +0300

    wifi: iwlwifi: mvm: cleanup beacon_inject_active during hw restart
    
    beacon_inject_active turns true and false via debugfs when we inject a
    beacon. During the time of the beacon injection we can't configure the
    FW with a beacon cmd. If we have a hw restart during the beacon injection
    then in the recovery flow we will not be able to configure the beacon
    cmd to the FW. Fix this by cleaning up this variable after an hw restart.
    
    Signed-off-by: Miri Korenblit <miriam.rachel.korenblit@intel.com>
    Signed-off-by: Gregory Greenman <gregory.greenman@intel.com>
    Link: https://lore.kernel.org/r/20230413213309.b021bce5e162.Ia5a0a0b5d8734f63077ceaac936176a345f8d3d1@changeid
    Signed-off-by: Johannes Berg <johannes.berg@intel.com>

diff --git a/drivers/net/wireless/intel/iwlwifi/mvm/mac80211.c b/drivers/net/wireless/intel/iwlwifi/mvm/mac80211.c
index ac5c4799fa7e..d6f29aa2a66f 100644
--- a/drivers/net/wireless/intel/iwlwifi/mvm/mac80211.c
+++ b/drivers/net/wireless/intel/iwlwifi/mvm/mac80211.c
@@ -1065,6 +1065,9 @@ static void iwl_mvm_restart_cleanup(struct iwl_mvm *mvm)
 	mvm->rx_ba_sessions = 0;
 	mvm->fwrt.dump.conf = FW_DBG_INVALID;
 	mvm->monitor_on = false;
+#ifdef CONFIG_IWLWIFI_DEBUGFS
+	mvm->beacon_inject_active = false;
+#endif
 
 	/* keep statistics ticking */
 	iwl_mvm_accu_radio_stats(mvm);